server/src/main/java/org/elasticsearch/cluster/metadata/MetadataDataStreamsService.java

```diff
@@ -199,23 +199,7 @@ static ClusterState updateDataLifecycle(
         Metadata.Builder builder = Metadata.builder(metadata);
         for (var dataStreamName : dataStreamNames) {
             var dataStream = validateDataStream(metadata, dataStreamName);
-            builder.put(
-                new DataStream(
-                    dataStream.getName(),
-                    dataStream.getIndices(),
-                    dataStream.getGeneration(),
-                    dataStream.getMetadata(),
-                    dataStream.isHidden(),
-                    dataStream.isReplicated(),
-                    dataStream.isSystem(),
-                    dataStream.isAllowCustomRouting(),
-                    dataStream.getIndexMode(),
-                    lifecycle,
-                    dataStream.isFailureStore(),
-                    dataStream.getFailureIndices(),
-                    dataStream.getAutoShardingEvent()
-                )
-            );
+            builder.put(new DataStream.Builder(dataStream).setLifecycle(lifecycle).setRolloverOnWrite(false).build());
```

Comment:
Hmm, not related to this specific PR, and maybe someone else can clarify, but it seems like we're very regularly clearing the rollover on write flag. Is that intentional? I can see it making sense in a lot of the other places, but I wouldn't have expected it being cleared when updating a lifecycle.

Reply:
Discussed and addressed in the other comment, so resolving this one.